### PR TITLE
Feat: update meeting location to firebase

### PR DESCRIPTION
**Pull request by QuocTrungTran**

### CODE REVIEW

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/MapDialog.java [hunk 1]
```diff
@@ -168,10 +169,13 @@ public void onClick(View view) {
...
-                Toast.makeText(context, getString(R.string.accepted_request), Toast.LENGTH_SHORT).show();
-                vm.acceptRequest(position);
-                dialog.dismiss();
+                if (meetingLocation != null) {
+                    // accept the request only when user specifies the location of meeting
+                    Toast.makeText(context, getString(R.string.accepted_request), Toast.LENGTH_SHORT).show();
+                    vm.acceptRequest(position, meetingLocation);
+                    dialog.dismiss();
+                }
+                Toast.makeText(context, getString(R.string.pick_location), Toast.LENGTH_SHORT).show();
```
Comment (zhpeng811) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/MapDialog.java:178]: this Toast will show up even if there's a meeting location? The logic here seems like it also I don't see it in the UI

Comment (QuocTrungTran) [reply app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/MapDialog.java:178]: Yeah my bad, the Toast should be in the condition when meetingLocation == null.

**Review by zhpeng811 — APPROVED** [review]
lgtm

File: app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java [hunk 2]
```diff
@@ -11,18 +12,29 @@
     private String requester;
     private String bookId;
     private REQUEST_STATUS status;
+    private LatLng meetingLocation;
```
Comment (kdvuong) [inline app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java:15]: GeoPoint class here for meetingLocation

**Review by c25vdw — APPROVED** [review]
lgtm

**Pull request merged**